sample/SampleApp/SampleApp.cpp

```diff
@@ -0,0 +1,118 @@
+// ConsoleApplication1.cpp : Defines the entry point for the console application.
+//
+
+#include <windows.h>
+#include <string>
+#include <assert.h>
+#include <playfab_c/PlayFabClientApi_c.h>
+
+PlayFabStateHandle stateHandle{ nullptr };
+PlayFabUserHandle userHandle{ nullptr };
+
+int main()
+{
+    HRESULT hr = PlayFabInitialize(&stateHandle);
+    assert(SUCCEEDED(hr));
+
+    {
+        PlayFabLoginWithCustomIDRequest request{};
+        request.createAccount = false;
+        request.customId = "GettingStartedGuide";
+        request.titleId = "E18D7";
+
+        XAsyncBlock async{};
+        async.callback = [](XAsyncBlock* async)
+        {
+            PlayFabResultHandle resultHandle{ nullptr };
+            PlayFabLoginResult* loginResult{ nullptr };
+            HRESULT hr = PlayFabLoginWithCustomIDGetResult(async, &resultHandle, &loginResult);
+            if (SUCCEEDED(hr))
+            {
+                hr = PlayFabUserDuplicateHandle(loginResult->userHandle, &userHandle);

```

Comment:
Requiring the title to call duplicate for the user handle here seems like it will be error prone, also it's not clear when/where loginResult is cleaned up. What if we did this instead:

struct PlayFabLoginResult
{
    PlayFabUserHandle userHandle;
    time_t lastLoginTime; // time_t{0} would be the equivalent of what nullptr would presviously represent
    bool newlyCreated;
};

HRESULT PlayFabLoginWithCustomIDGetResult(
    _In_ XAsyncBlock* async,
    _Out_ PlayFabResultHandle* handle,
    _Out_ PlayFabLoginResult* result // Title provides the memory for the returned LoginResult
) noexcept;

 PlayFabLoginResult loginResult{ }; // Stack allocated. Title is required to call close handle on the user, but doesn't have to duplicate the handle to retain the user.
 HRESULT hr = PlayFabLoginWithCustomIDGetResult(async, &resultHandle, &loginResult);

Reply:
Another option would be accessor methods get the user and other fields so the we can make the duplicate call behind the scenes that the developer doesn't have to think about that.